client/spring-zeebe/src/main/java/io/camunda/zeebe/spring/client/config/processor/DeploymentPostProcessor.java

```diff
@@ -40,9 +53,25 @@ public boolean test(final ClassInfo beanInfo) {
       DeployProcessCommandStep1 deployWorkflowCommand = client
         .newDeployCommand();
 
-      DeploymentEvent deploymentResult = value.getClassPathResources()
+      List<Resource> locations = Stream.of(Optional.ofNullable(value.getBpmnLocations()).orElse(Collections.emptyList()))
+        .flatMap(List::stream)
+        .flatMap(location -> Stream.of(getResources(location)))
+        .collect(Collectors.toList());
+
+      Stream<DeployProcessCommandStep1.DeployProcessCommandBuilderStep2> deployProcessCommandBuilderStep2StreamOfLocations = locations.stream().map(resource -> {
+        try {
+          return deployWorkflowCommand.addResourceStream(resource.getInputStream(), resource.getFilename());
+        } catch (IOException e) {
+        }
+        return null;
+      }).filter(Objects::nonNull);
+
+      Stream<DeployProcessCommandStep1.DeployProcessCommandBuilderStep2> deployProcessCommandBuilderStep2StreamOfResources = value.getClassPathResources()
```

Comment:
If we condense everything into one attribute - we don't need duplicated code here :-)